internal/stopwatch/timed.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package stopwatch
+
+import (
+	"go.uber.org/atomic"
+	"time"
+)
+
+// Timed runs the given function and adds the time it took to the given atomic.Int64. Thread-safe
+func Timed(timer *atomic.Int64, f func()) {
```

Comment:
This is a place where you want to be specific that the timer is nanos... This function being exported is also a bit weird... especially since this package provides the `Stopwatch` which is supposed to make this redundant?